dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/OperationParamUtils.java

```diff
@@ -138,35 +137,6 @@ else if (filters.length == 4) {
     return queryItem;
   }
 
-  /**
-   * Creates a QueryFilter from the given query string. Query is on format
-   * {operator}:{filter-value}. Only the filter-value is mandatory. The EQ QueryOperator is used as
-   * operator if not specified. We split the query at the first delimiter, so the filter value can
-   * be any sequence of characters
-   *
-   * @throws BadRequestException given invalid query string
-   */
-  public static QueryFilter parseQueryFilter(String filter) throws BadRequestException {
-    if (StringUtils.isEmpty(filter)) {
-      return null;
-    }
-
-    if (!filter.contains(DimensionalObject.DIMENSION_NAME_SEP)) {
-      return new QueryFilter(QueryOperator.EQ, filter);
-    }
-
-    return operatorValueQueryFilter(FILTER_ITEM_SPLIT.split(filter), filter);
-  }
-
-  private static QueryFilter operatorValueQueryFilter(String[] operatorValue, String filter)
-      throws BadRequestException {
-    if (null == operatorValue || operatorValue.length < 2) {
-      throw new BadRequestException("Query item or filter is invalid: " + filter);
-    }
-
-    return operatorValueQueryFilter(operatorValue[0], operatorValue[1], filter);
-  }
-
   public static QueryFilter operatorValueQueryFilter(String operator, String value, String filter)
```

Comment:
When I worked on harmonizing order in /events I moved parsing of filters from the service layer (if I remember correctly) to the web layer https://github.com/dhis2/dhis2-core/blob/6b058254b83cfab4f92a3f26e9b5a11bbe95199f/dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/RequestParamsValidator.java#L251 I wanted to do the same for /trackedEntities but could not as the interaction between order, attributes and filters was so complex. I mention this so we don't forget to look at the changes for events and how to do the same for trackedEntities 😅